src/sonar_widget/SonarPlot.cc

```diff
@@ -142,3 +148,63 @@ void SonarPlot::rangeChanged(int value)
     range = value;
     drawOverlay();
 }
+
+void SonarPlot::sonarPaletteChanged(int index){
+    colorMapID = (Colormap) index;
+}
+
+void SonarPlot::generateBearingTable(int beams, base::Angle bhoriz, base::Angle bvert) {
+    bearingTable.clear();
+
+    double factor;
+    bhoriz.getDeg() == 120.0 && bvert.getDeg() == 20.0 ? factor = 0.86602540 : factor = 1.0;
+
+    for (int i = 0; i < beams; i++) {
+        double rad = asin(((2 * i - beams) * 1.0 / beams) * factor);
+        const base::Angle new_angle = base::Angle::fromRad(rad);
+        bearingTable.push_back(new_angle);
+
+    }
+}
+
+void SonarPlot::generateColormaps() {
```

Comment:
a) this package depends on the frame-helper which depends on opencv, so no new dep would be added
b) if you are unhappy which the opencv color-sceme please add this color map to the frame_helper and make the colormap maybe configureable (so that also the opencv could be used). This would enable the use of this color map outside this class.